src/interactive/Dropdown.jsx

```diff
@@ -106,18 +107,37 @@ class Dropdown extends React.PureComponent<Props, State> {
 		}
 	};
 
-	onBodyClick = (e: SyntheticMouseEvent<*>) => {
-		if (this.props.closeOnBodyClick) {
-			this.closeContent(e);
-			return;
+	/** Handles execution of menuItem action and then closes the dropdown
+	 	_if_ `closeOnMenuItemClick` is set to true and there are menuItems.
+	 	If there are menuItems but `closeOnMenuItemClick` is false it'll just
+	 	fire the action and keep the dropdown open. */
+	onDropdownMenuItemClick = (
+		e: SyntheticMouseEvent<*>,
+		itemProps: { onClick?: (SyntheticMouseEvent<*>) => void }
```

Comment:
We don't need the entire object of item props here since `onClick` is the only property used